midi_app_controller/state/state_manager.py

```diff
@@ -269,8 +284,26 @@ def start_handling(self) -> None:
             midi_out=self._midi_out,
         )
 
+    def select_recent_midi_ports(self):
+        """Select MIDI ports that were recently used with the current controller."""
+        if (
+            self.selected_controller
+            and self.selected_controller.path in self.recent_midi_ports_for_controller
+        ):
+            ports = self.recent_midi_ports_for_controller[self.selected_controller.path]
+            if ports["in"] in self.get_available_midi_in():
+                self.selected_midi_in = ports["in"]
+            if ports["out"] in self.get_available_midi_out():
+                self.selected_midi_out = ports["out"]
+
     def save_state(self):
         """Saves the current settings to the disk."""
+        if self.selected_controller is not None:
+            self.recent_midi_ports_for_controller[self.selected_controller.path] = {
+                "in": self.selected_midi_in,
+                "out": self.selected_midi_out,
+            }
+
```

Comment:
It can be deleted as it doesn't change anything. The dict was already updated in `select_midi_out`/`select_midi_in`.